google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -401,6 +401,9 @@ public boolean isClosed() {
 
   @Override
   public void setAutocommit(boolean autocommit) {
+    if (isAutocommit() == autocommit) {
```

Comment:
Move this `if` block to after the line with `ConnectionPreconditions.checkState(!isClosed(), CLOSED_ERROR_MSG);`. Calling this method, even if the value is the same as now, should throw an exception. The current implementation will actually do that as well, as the `isAutocommit()` call will check it, but that is not clear from the current implementation, so for readability it would be better to move this a bit down.

Reply:
Fixed